tests/integration/test_upgrade_charm.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import get_config_values
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+app_name = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_from_edge_and_upgrade_from_local_path(ops_test, charm_under_test):
+    """Deploy from charmhub and then upgrade with the charm-under-test."""
+    logger.info("deploy charm from charmhub")
+    resources = {"unused-image": METADATA["resources"]["unused-image"]["upstream-source"]}
+    await ops_test.model.deploy(f"ch:{app_name}", application_name=app_name, channel="edge")
+
+    config = {"scrape_interval": "15s", "scrape_timeout": "10s"}
```

Comment:
It is not obvious why a upgrade test needs to set these config values ?

Reply:
added doc.

Reply:
Renamed test to `test_config_values_are_retained_after_pod_upgraded`.